### PR TITLE
parsing fixes, used for kr run

### PR DESCRIPTION
**Pull request by bazinski**
- fix 1 remaining parsing warning for mc2raw
- handles multiple end of digit markers correctly
- remove error and warning messages, see histograms and eventually qc for those messages.
- 2d histograms for parsing errors, rather illuminating
- fix corrupted halfchamber header where additional header words and data version are zero now.
- command line option to set DigitHCHeader.numberHCW --hcheaderwords
- command line option to set data version --hcheadermajor  only major version is used.
- command line option to expressly ignore halfchamber header (location, major version and addtional words blank or identical)
- histograms neatened up
- tracklets at the position of the stuck half chamber header are still there, bit corruption, so tracklets at hcid 512, row 8 are to be ignored, remove the spike
- if a digit mcmheader is corrupt digit data is dumped, there is however a possibilty to recover digits later on in the buffer of that rob, needs more investigation for producing corrupt digits.
- stats messages to qc are stripped out of this PR

This is the hash used for Kr runs.

I have not been through all the permutations of combinations there are too many now, the ones to process previous data work though.

### CODE REVIEW

**Comment by bazinski** [comment]
@shahor02 @tdietel can this be merged, i dont know what the deal is with o2-cs8 and macOS (who uses mac??) but this has been stuck since thursday.

**Comment by shahor02** [comment]
Some people use mac... And it is pickier on some code aspects than others. Anyway, if you have no chance to test it privately on mac, will merge it once @tdietel approves it.

**Comment by bazinski** [comment]
@shahor02 the mac comment as a dig at @tdietel ;-)

**Review by tdietel — DISMISSED** [review]
The changes look reasonable - although I only glanced over the. Let's go ahead with the merge.

**Comment by bazinski** [comment]
@shahor02 @tdietel dont merge this, it crashes for 3 timeframes in 600k, investigating. Well something crashes, not sure where.

**Comment by bazinski** [comment]
parsing error now fixed for the combination of cli options we are using for corrupted data and the instance where we receive the entire link data of 27 256 bit lines of 
0x0 0x0 0xeeeeeeee 0xeeeeeeee 0xeeeeeeee 0xeeeeeeee 0xeeeeeeee 0xeeeeeeee
The combination of cli options did not take into account skipping the half chamber header when half chamber header is to be ignored. Now fixed.

**Comment by bazinski** [comment]
@shahor02 @tdietel can this be merged now, goes happily through the Kr data. All the problems converting the data are external to this code. Also works fine for the mc2raw.

**Comment by tdietel** [comment]
Then we should merge it.

If this is the version we use to read the krypton data, we should also note it down somewhere. Maybe in a Jira ticket, or as a git tag, or both. What do you think?

**Review by tdietel — APPROVED** [review]

**Comment by bazinski** [comment]
ja will put the git hash once merged in the jira ticket

**Comment by bazinski** [comment]
@shahor02 can this be merged?

**Comment by bazinski** [comment]
@shahor02 this one

**Pull request merged**
